engines/instance_verification/lib/instance_verification/engine.rb

```diff
@@ -99,7 +99,10 @@ def self.verify_instance(request, logger, system)
     cache_params = {}
     # we are checking the base product so we pick the first registration code
     # PAYG instances have no registration code
-    cache_params = { token: Base64.decode64(system.pubcloud_reg_code.split(',')[0]), instance_data: decoded_instance_data } unless system.payg?
+    unless system.payg? || system.pubcloud_reg_code.blank?
```

Comment:
When payg systems don't have a regcode, wouldn't it be the same to only check: 

```suggestion
    unless system.pubcloud_reg_code.blank?
```